packages/core/package.json

```
@@ -2,15 +2,17 @@
   "private": true,
   "name": "@aws-amplify/ui-core",
   "version": "0.0.1",
-  "module": "index.ts",
-  "main": "dist/index.js",
+  "type": "module",
+  "main": "dist/index.cjs",
+  "module": "dist/index.js",
   "types": "dist/index.d.ts",
   "files": [
-    "dist/*.{js,ts}",
-    "dist/**/*.{js,ts}"
+    "dist"
   ],
   "scripts": {
-    "build": "tsc",
+    "prebuild": "rimraf dist",
```

Should we have a separate `yarn clean` script here? I find explicit clean scripts when I'm looking at module/build issues.

```suggestion
    "clean": "rimraf dist",
    "prebuild": "yarn clean",
```

---

You know, I went back-and-forth on this.  What I noticed was that it was tough to trace when something was being used and where. (`build:icons` is an example of this)

Having `clean` explicitly defined kinda felt like I was promoting it to a public API (e.g. `yarn react clean`) vs. having only `build|dev` being the public APIs.

I hope the indirection from `build => prebuild => clean` is clearer with `build => prebuild` 🙏 

(I don't have a strong opinion – I've done NPM scripts every which way since [per-env](https://github.com/ericclemmons/per-env))

---

Is there a use-case to run _only_ a clean? I've realized whenever I'm doing a clean it's because I want to blow up the build artifacts and rebuild them, so just being part of `prebuild` makes sense to me. But if there are use cases for "clean and not build" I'm for it.

---

Great question. I don't have one.  Your use-case models my experience as well:

- "Build for production", which means a clean slate so we don't re-publish old artifacts.
- "Dev" _should not_ remove clean because I found out it breaks hot-reload when those references disappear.

I suppose there could be a reason to clean and do dev, but IME `yarn install` is basically our starting point for "setup and build everything, and I'll chose what part I'm working on".